pyarn/lockfile.py

```diff
@@ -131,12 +126,20 @@ def _dump_keyval(key, value, outfile, indent_level):
         outfile.write(' ')
         if isinstance(value, str):
             # Always quote string values
-            # TODO: see quoting keys
+            # TODO: use json.dump to quote the value instead
+            #   (the lexer would also have to interpret strings using json.load)
             outfile.write(f'"{value}"')
         else:
             json.dump(value, outfile)
         outfile.write('\n')
 
 
+def _quote_key_if_needed(key):
+    # The key may be a comma-separated list of keys
+    keys = map(str.strip, key.split(","))
```

Comment:
There is another potential problem here:
```
"foo@https://quote.in.url/?x=y,z"
```
With this change, this becomes
```
"foo@https://quote.in.url/?x=y", "z":
```

There is no way to distinguish between comma as a separator vs. comma as part of a string with the current parser. IMO, not splitting commas is much more likely to cause issues then corner cases such as this, WDYT?

Reply:
I agree with your opinion here. Splitting might be a better approach however I'm wondering what would happen if we were to encounter such corner case? Would it cause incorrect values or failure? (Pardon my lack of knowledge regarding the context this is being used)

Reply:
That is a good question. I realized this change needs a lot more context, added in https://github.com/containerbuildsystem/pyarn/pull/5#issuecomment-768143402